WebView/ViewController.swift

```diff
@@ -20,25 +22,71 @@ class ViewController: UIViewController, WKNavigationDelegate {
+        
+        progressView = UIProgressView(progressViewStyle: .default)
+        progressView.sizeToFit()
+        let progressButton = UIBarButtonItem(customView: progressView)
+        
+        toolbarItems = [progressButton, spacer, back, forward, refresh]
+        navigationController?.isToolbarHidden = false
+        
+        webView.addObserver(self, forKeyPath: #keyPath(WKWebView.estimatedProgress), options: .new, context: nil)
+        
+        
+        guard let url = URL(string: "https://" + webSites[0]) else { return }
         webView.load(URLRequest(url: url))
         webView.allowsBackForwardNavigationGestures = true
     }
+    
     @objc func openTapped() {
-        let ac = UIAlertController(title: "Open page...", message: nil, preferredStyle: .actionSheet)
-        ac.addAction(UIAlertAction(title: "apple.com", style: .default, handler: openPage))
-        ac.addAction(UIAlertAction(title: "google.com", style: .default, handler: openPage))
-        ac.addAction(UIAlertAction(title: "Cancel", style: .cancel))
-        ac.popoverPresentationController?.barButtonItem = navigationItem.rightBarButtonItem
-        present(ac, animated: true)
+        let alertController = UIAlertController(title: "Open page...", message: nil, preferredStyle: .actionSheet)
+        
+        for website in webSites {
+            alertController.addAction(UIAlertAction(title: website, style: .default, handler: openPage))
+        }
+        
+        
+        alertController.addAction(UIAlertAction(title: "Cancel", style: .cancel))
+        alertController.popoverPresentationController?.barButtonItem = navigationItem.rightBarButtonItem
+        present(alertController, animated: true)
     }
-    func openPage(action: UIAlertAction) {
-        guard let actionTitle = action.title else { return }
-        guard let url = URL(string: "https://" + actionTitle) else { return }
+    
+    private func openPage(action: UIAlertAction) {
+        guard let actionTitle = action.title, let url = URL(string: "https://" + actionTitle) else { return }
         webView.load(URLRequest(url:url))
     }
+    
     func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
         title = webView.title
     }
+    
+    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
+        if keyPath == "estimatedProgress"{
+            progressView.progress = Float(webView.estimatedProgress)
+        }
+    }
+    
+    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
+        let url = navigationAction.request.url
+        
+        if let host = url?.host {
+            
+            if host.contains(webSites[1]) {
+                decisionHandler(.allow)
+                return
+            }else {
```

Comment:
minor styling
```suggestion
            } else {
```

Reply:
That was an undetected nonintentional style! 
Fixing!